### PR TITLE
Creating RTOS-specific tutorials from common sections.

### PR DESCRIPTION
**Pull request by ralph-lange**
- Factored out common sections into `_includes/first_application_rtos_common`. (Unfortunately, `include_relative` macro with includes from current folder did not work - seems to be a bug.)
- Created overview page which asks reader to decide for one RTOS.
- Removed specific page ` tutorials/advanced/zephyr/zephyr_getting_started` but created redirect instead.

### CODE REVIEW

**Comment by ralph-lange** [comment]
@pablogs9, as far as I see, you've made a couple of improvements to the Zephyr-specific getting started page. May I ask you kindly, to integrate these improvements in the new structure so that the versions for NuttX and FreeRTOS also benefit from them?

**Comment by pablogs9** [comment]
Hi @ralph-lange, sorry for the delay. Some points I'm thinking about:

- What about removing this line `(The created workspace is platform- and RTOS-specific. Use `freertos` or `zephyr` instead of `nuttx` to choose FreeRTOS or Zephyr, respectively.)` since in the landing page you have selected the desired RTOS. Or maybe is it ok to remind that the `firmware` is platform specific. What do you think?

- In the table in line 23 in `nuttx.md`, what about leaving only the rows regarding the specific RTOS? Same in table in `section_04_demo_description.md`.

I have implemented some changes in the last commit. Do you have some more in mind? Anyways, @jamoralp is going to review all the micro-ROS procedures (and hopefully test all RTOSes and boards) this week, so maybe he can do some contributions to the tutorials while he is replicating them.

**Comment by ralph-lange** [comment]
Hi @pablogs9, I just removed the sentence `(The created workspace is platform- and RTOS-specific. Use freertos or zephyr instead ofnuttx to choose FreeRTOS or Zephyr, respectively.)` from all three versions of the tutorial, cf. latest commits. This was really redundant information.
However, I would like to keep the table with the links to all three versions of the main app code, to allow the reader verifying our statement that the code really differs in the signature of the main function and some includes only. For the same argument, I would keep the table in section_04_demo_description.md. It does not take much space (and thus will not distract the reader), but illustrates again that micro-ROS is very much independent of the underlying RTOS and that there are only small differences between the RTOSes in setting up a new application.

File: _docs/tutorials/core/first_application_rtos/nuttx.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+```bash
+# Create step
+ros2 run micro_ros_setup create_firmware_ws.sh nuttx olimex-stm32-e407
+```
+
+{% include first_application_rtos_common/section_03_configuring_firmware.md %}
+
+|  RTOS        | `[APP]`         | `[OPTIONS]`                  |                                  Configured app                                  |
+| :------: | --------------- | ---------------------------- | :------------------------------------------------------------------------------: |
+|  **NuttX**   | **`uros_pingpong`** |                              | [**Source**](https://github.com/micro-ROS/apps/tree/dashing/examples/uros_pingpong)  |
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_rtos/nuttx.md:23]:
```suggestion
|  **NuttX**   | **`uros_pingpong`** |                              | [**Source**](https://github.com/micro-ROS/apps/tree/feature/pingpong_app/examples/uros_pingpong)  |
```

**Pull request merged**